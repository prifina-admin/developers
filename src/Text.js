import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { space, typography } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";

const customPropsList = { bold: "fontWeight", italic: "fontStyle" };

const customProps = props => {
  //console.log("PROPS CUSTOM ", props);
  let propsList = [];
  Object.keys(props).forEach(p => {
    if (Object.keys(customPropsList).indexOf(p) > -1) {
      let cssProp = {};
      if (p === "bold") {
        cssProp[customPropsList[p]] = props.theme.fontWeights["bold"];
      } else {
        cssProp[customPropsList[p]] = p;
      }
      propsList.push(cssProp);
    }
  });
  return propsList;
};
/*
//https://styled-system.com/how-it-works
const TextElement = styled.div`
  ${textStyle}
  ${typography}
  ${space}
  ${color}
  ${customProps}
  color: ${props => (props.color ? props.color : "currentColor")};
  font-family: ${props => (props.fontFamily ? props.fontFamily : "body")};
`;
*/
/*
//https://styled-components.com/releases
const TextElement = styled("div").withConfig({
  shouldForwardProp: (prop, defaultValidatorFn) =>
    !["color", "fontFamily"].includes(prop),
})`
  ${textStyle}
  ${typography}
  ${space}
  ${color}
  ${customProps}
`;
*/
const color = props => {
  console.log("color PROPS ", props);

  return {
    color: props.color ? props.color : props.theme.colors["textPrimary"],
  };
};
const textStyles = props => {
  return props.textStyle ? props.theme.textStyles[props.textStyle] : null;
};
const colorStyles = props => {
  return props.colorStyle ? props.theme.colorStyles[props.colorStyle] : null;
};
// https://styled-system.com/guides/build-a-box

const TextElement = styled.div(
  typography,
  space,
  color,
  customProps,
  textStyles,
  colorStyles,
);

const Text = ({ as, ...props }) => {
  console.log("PROPS ", props);
  //const {colors} = useTheme();
  //console.log(colors);
  /* color: themeGet("palette.primary.main")(props), */
  return <TextElement as={as} {...props} />;
};

Text.displayName = "Text";
Text.defaultProps = {
  as: "div",
};

Text.propTypes = {
  ...styledProps.space,
  ...styledProps.typography,
  ...styledProps.color,
  bold: PropTypes.bool,
  italic: PropTypes.bool,
};

export default Text;
