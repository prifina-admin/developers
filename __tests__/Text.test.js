import React from "react";
import renderer from "react-test-renderer";
import Text from "../src/Text";
import ThemeProvider from "../src/theme/ThemeProvider";

test("Text renders correctly", () => {
    const tree = renderer.create(<ThemeProvider>
        <Text> Something{" "} </Text>
    </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Text renders fontStyle correctly", () => {
    const tree2 = renderer.create( <ThemeProvider><Text italic> Something </Text></ThemeProvider> ).toJSON();
    expect(tree2).toMatchSnapshot();
});
