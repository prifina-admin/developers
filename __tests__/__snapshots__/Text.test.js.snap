// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <div
    className="Text__TextElement-sc-102lho-0 glCa-di"
  >
     Something
     
     
  </div>
</div>
`;

exports[`Text renders fontStyle correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <div
    className="Text__TextElement-sc-102lho-0 jBaQIo"
  >
     Something 
  </div>
</div>
`;
